libp2p/protocols/connectivity/autorelay.nim

```diff
@@ -0,0 +1,110 @@
+# Nim-LibP2P
+# Copyright (c) 2022 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import chronos, chronicles, times, tables, sequtils, options
+import ../../switch,
+       relay/[client, utils]
+
+logScope:
+  topics = "libp2p autorelay"
+
+type
+  OnReservationHandler = proc (ma: MultiAddress): Future[void] {.gcsafe, raises: [Defect].}
+
+  AutoRelay* = ref object of Service
+    running: bool
+    runner: Future[void]
+    client: RelayClient
+    npeers: int
+    peers: Table[PeerId, Future[void]]
```

Comment:
Can we have a more descriptive name here? Maybe `relayPeers` if I understood it correctly.